test/jdk/java/awt/EtchedBorder/ScaledEtchedBorderTest.java

```diff
@@ -0,0 +1,249 @@
+                    checkShadow = false;
+                    thickness = 0;
+                }
+            }
+        }
+    }
+
+    private static void verifyThickness(int x, int y, int thickness, double scaling, String orientation) {
+        int expected = (int) Math.floor(scaling);
+        if (thickness != expected) {
+            throw new RuntimeException("Unexpected " + orientation + " Border thickness at x:"
+                                       + x + " y: " + y + ". Expected: " + expected + " Actual: " + thickness);
+        }
+    }
+
+    private static void checkVerticalBorder(int x, BufferedImage img, double scaling) {
+        int thickness = 0;
+        boolean checkShadow = false;
+        boolean checkHighlight = false;
+        for (int y = 0; y < img.getHeight(); y++) {
+            int color = img.getRGB(x, y);
+            if (!checkHighlight && !checkShadow) {
+                if (color == shadow.getRGB()) {
+                    checkHighlight = true;
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    throw new RuntimeException("Vertical Border was clipped or overdrawn.");
+                }
+            } else if (checkHighlight) {
+                if (color == shadow.getRGB()) {
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    verifyThickness(x, y, thickness, scaling, "Vertical");
+                    checkHighlight = false;
+                    checkShadow = true;
+                    thickness = 1;
+                } else {
+                    throw new RuntimeException("Vertical Border has empty space between highlight and shadow.");
+                }
+            } else {
+                if (color == shadow.getRGB()) {
+                    throw new RuntimeException("Border colors reversed.");
+                } else if (color == highlight.getRGB()) {
+                    thickness++;
+                } else {
+                    verifyThickness(x, y, thickness, scaling, "Vertical");
+                    checkShadow = false;
+                    thickness = 0;
+                }
+            }
+        }
+    }
+
+    private static void createGUI() {
+        // Render content panel
+        contentPanel = new JPanel();
+        contentPanel.setLayout(new BoxLayout(contentPanel, BoxLayout.Y_AXIS));
+
+        for (int i = 0; i < 4; i++) {
+            parentPanel = new JPanel(new BorderLayout());
+            parentPanel.setBorder(BorderFactory.createEmptyBorder(0, i, 4, 4));
+
+            childPanel = new JPanel(new BorderLayout());
+            childPanel.setBorder(BorderFactory.createEtchedBorder(highlight, shadow));
+            childPanel.add(Box.createRigidArea(SIZE), BorderLayout.CENTER);
+
+            parentPanel.add(childPanel, BorderLayout.CENTER);
+            contentPanel.add(parentPanel);
+        }
+
+        frame = new JFrame("Swing Test");
```

Comment:
Don't you want to go frameless? It would make the test headless. In [this comment](https://github.com/openjdk/jdk/pull/7449#discussion_r878130575), I provided [sample code](https://github.com/alisenchung/jdk/compare/8279614...aivanov-jdk:alisen-8279614-titledBorder) which works without creating frame: just panels and borders.